### PR TITLE
DragonFlyBSD verification for ioctl API

### PR DESCRIPTION
**Pull request by kotylamichal**

### CODE REVIEW

File: dragonflybsd/esrt-api.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Instructions for getting user-space ESRT API on DragonFlyBSD
+
+Read [README.md](./README.md) first for general information.
+
+The plan is:
+ * obtain `ioctl-efitool` branch of https://github.com/3mdeb/DragonFlyBSD/tree/esrt
```
Comment (tomzy-0) [inline dragonflybsd/esrt-api.md]: @kotylamichal lets point to the correct branch `https://github.com/3mdeb/DragonFlyBSD/tree/ioctl-efitool`

File: dragonflybsd/esrt-api.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+```
+# cd /usr
+# make src-create-shallow
+# make src-update
+```
+
+Sources initialization may take a while. After that, add 3mdeb repository:
+
+```
+# git remote add 3mdeb https://github.com/3mdeb/DragonFlyBSD.git
```
Comment (tomzy-0) [inline dragonflybsd/esrt-api.md:27]: @kotylamichal cd `/usr/src` is missing

File: dragonflybsd/esrt-api.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+## Kernel compilation and installation
+
+```
+# cd /usr/src/sys/config
+# mkdir /root/kernels
+# cp X86_64_GENERIC /root/kernels/MYKERNEL
+# ln -s /root/kernels/MYKERNEL
+
+# make -j4 nativekernel KERNCONF=MYKERNEL
```
Comment (tomzy-0) [inline dragonflybsd/esrt-api.md]: @kotylamichal looks like make nativekernel should be executed from /usr/src so I propose here to run `make -j 4 -C ../../ nativekernel KERNCONF=MYKERNEL

File: dragonflybsd/esrt-api.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+# make installkernel KERNCONF=MYKERNEL
```
Comment (tomzy-0) [inline dragonflybsd/esrt-api.md]: @kotylamichal same here `make -C ../../ KERNCONF=MYKERNEL`

File: dragonflybsd/esrt-api.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+## EFI bootloader compilation and installation
+
+```
+# cd /usr/src/stand/boot/efi/libefi
+# make
+
+# cd /usr/src/stand/boot/efi/loader
+# make
```
Comment (tomzy-0) [inline dragonflybsd/esrt-api.md:48]: @kotylamichal this returns lots of undefined references

Comment (tomzy-0) [reply dragonflybsd/esrt-api.md:48]: @kotylamichal needed to go into `cd /usr/src/stand` run make there (which returns errors) and then could compile loader

File: dragonflybsd/esrt-api.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+## Verification
+
+Kernel and bootloader from 3mdeb sources should be installed which was described
+in [Building custom kernel](#Building-custom-kernel) section. To verify ESRT
+tables functionality you need to load EFI runtime module:
+
+```
+# kldload efirt
```
Comment (tomzy-0) [inline dragonflybsd/esrt-api.md:60]: @kotylamichal just loading the module should not print the esrt table, we should remove that from code

File: dragonflybsd/esrt-api.md (hunk 5 above)
Comment (tomzy-0) [reply dragonflybsd/esrt-api.md:48]: This should be fixed, maybe we just need to compile something more before the loader?

**Review by tomzy-0 — CHANGES_REQUESTED** [review]

File: dragonflybsd/esrt-api.md (hunk 1 above)
Comment (kotylamichal) [reply dragonflybsd/esrt-api.md]: done

File: dragonflybsd/esrt-api.md (hunk 2 above)
Comment (kotylamichal) [reply dragonflybsd/esrt-api.md:27]: done

File: dragonflybsd/esrt-api.md (hunk 3 above)
Comment (kotylamichal) [reply dragonflybsd/esrt-api.md]: done

File: dragonflybsd/esrt-api.md (hunk 4 above)
Comment (kotylamichal) [reply dragonflybsd/esrt-api.md]: done

File: dragonflybsd/esrt-api.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
+# Instructions for getting user-space ESRT API on DragonFlyBSD
+
+Read [README.md](./README.md) first for general information.
+
+The plan is:
+ * obtain `ioctl-efitool` branch of
+ https://github.com/3mdeb/DragonFlyBSD/tree/octl-efitool fork of DragonFlyBSD
```
Comment (tomzy-0) [inline dragonflybsd/esrt-api.md]:
```suggestion
 https://github.com/3mdeb/DragonFlyBSD/tree/ioctl-efitool fork of DragonFlyBSD
```

File: dragonflybsd/esrt-api.md (hunk 6 above)
Comment (kotylamichal) [reply dragonflybsd/esrt-api.md:60]: @TomaszAIR removed in last commit: https://github.com/3mdeb/DragonFlyBSD/pull/6/commits/571b1c18ff8926f95b7302de4e6af14d336f2420

Comment (tomzy-0) [reply dragonflybsd/esrt-api.md:60]: Yes, table is not printed now

File: dragonflybsd/esrt-api.md (hunk 5 above)
Comment (tomzy-0) [reply dragonflybsd/esrt-api.md:48]: @kotylamichal worked without rebuilding bootloader so we can get rid off this section

File: dragonflybsd/esrt-api.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+# make install
+# reboot
+```
+
+## Prepare user-space tool
+
+```
+# cd /usr/src/usr.sbin/efitable
+# make
+# make install
```
Comment (tomzy-0) [inline dragonflybsd/esrt-api.md:49]: @kotylamichal lets show how to compile kernel then this tool and at the end reboot the system

Comment (kotylamichal) [reply dragonflybsd/esrt-api.md:49]: done

File: dragonflybsd/esrt-api.md (hunk 5 above)
Comment (kotylamichal) [reply dragonflybsd/esrt-api.md:48]: removed

**Review by tomzy-0 — APPROVED** [review]
works correctly

**Pull request merged**